src/hooks/login/useWalletConnectV2Login.ts

```diff
@@ -0,0 +1,243 @@
+import { useRef, useState } from 'react';
```

Comment:
I see that this hook is 80% the same with useWalletConnectLogin. Could you extract the common logic to avoid code duplication?

Reply:
We will keep the hooks separate for now since WC2 is still in a beta phase and more breaking changes might apprear. This way any changes won't affect WC1 that is already in a stable release.